lib/aws/proxy.ts

```diff
@@ -0,0 +1,84 @@
+import { Construct } from "constructs";
+import { Config } from "./config";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { AutoScalingGroup } from "aws-cdk-lib/aws-autoscaling";
+import { readFileSync } from "fs";
+import { ApplicationLoadBalancer } from "aws-cdk-lib/aws-elasticloadbalancingv2";
+
+// ProxySetup is a construct that sets up a proxy server in the VPC
+//
+// The proxy server, comprises of 1 ASG and 1 LB, the ASG runs a ec2 instance template with envoy proxy installed, the envoy proxy should point to the host
+// The load balancer should be internet facing and should be able to route traffic to the ASG instances
+export class ProxySetup extends Construct {
+  lb_sg: ec2.ISecurityGroup;
+  asg_sg: ec2.ISecurityGroup;
+  asg: AutoScalingGroup;
+  lb: ApplicationLoadBalancer;
+  constructor(
+    scope: Construct,
+    config: Config,
+    vpc: ec2.Vpc,
+    host: string,
+    host_sg: ec2.ISecurityGroup,
+  ) {
+    super(scope, "ProxySetup");
+
+    // read from envoy.sh 
+    // infer the host as UPSTREAM_HOST
+    const userData = ec2.UserData.custom(
+      readFileSync("lib/aws/proxy/envoy.sh", "utf8") +
+      "UPSTREAM_HOST=" + host + "\n" +
+      "CLUSTER_NAME=" + "hyperswitch-cluster" + "\n" +
```

Comment:
Is this cluster name represent eks cluster name? If yes, It is `hs-eks-cluster`.